### PR TITLE
popup: save tab on reload

### LINKED ISSUE #4809: Select the last opened tab when opening the popup
### Is your request related to a problem? Please describe.

The Messages tab in the popup is always selected by default, even if the user needs one of the other tabs more often.

### Describe the solution you'd like

Remember which tab was opened and select it again.

### PR DESCRIPTION
**Pull request by Weredime**
Resolves #4809

### Changes
Saves the current popup tab on reload.

### Tests
Firefox

### CODE REVIEW

**Review by WorldLanguages — DISMISSED** [review]
Let's use `chrome.storage.local` instead of `localStorage`.

File: webpages/popup/index.js [hunk 1]
```diff
@@ -86,7 +89,14 @@ chrome.runtime.sendMessage("getSettingsInfo", (res) => {
     _addonId: "__settings__",
   });
   vue.popups = popupObjects;
-  vue.setPopup(vue.popups[0]);
+  chrome.storage.local.get("lastSelectedPopup", ({ lastSelectedPopup }) => {
+    let id = 0;
+    if (lastSelectedPopup !== null) {
```
Comment (apple502j) [inline webpages/popup/index.js:94]:
```suggestion
    if (typeof lastSelectedPopup === "string") {
```

File: webpages/popup/index.js [hunk 2]
```diff
@@ -110,6 +120,9 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
       if (removeIndex !== -1) vue.popupsWithIframes.splice(removeIndex, 1);
       removeIndex = vue.popups.findIndex((popup) => popup._addonId === addonId);
       vue.popups.splice(removeIndex, 1);
+      if (vue.popups.indexOf(vue.currentPopup) === -1) {
```
Comment (apple502j) [inline webpages/popup/index.js:123]:
```suggestion
      if (!vue.popups.includes(vue.currentPopup)) {
```

**Review by apple502j — CHANGES_REQUESTED** [review]

**Review by mxmou — APPROVED** [review]

File: webpages/popup/index.js [hunk 3]
```diff
@@ -110,6 +120,9 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
       if (removeIndex !== -1) vue.popupsWithIframes.splice(removeIndex, 1);
       removeIndex = vue.popups.findIndex((popup) => popup._addonId === addonId);
       vue.popups.splice(removeIndex, 1);
+      if (!vue.popups.includes(vue.currentPopup)) {
+        vue.setPopup(vue.popups[0]); // set to default popup if current popup is no longer available
+      }
```
Comment (WorldLanguages) [inline webpages/popup/index.js:125]: This fixes a bug that happens on master branch. Good catch!

**Review by WorldLanguages — APPROVED** [review]

**Comment by WorldLanguages** [comment]
Hope this improves DX! As well as user experience of course hehe
@mxmou I bet before this, you had scratch-messaging and cloud-games disabled, just to open the addons tab in the popup by default :P I had to do that a lot too

File: webpages/popup/index.js (hunk 3 above)
Comment (TheColaber) [reply webpages/popup/index.js:125]: Oops

**Pull request merged**